lua/nvim-treesitter.lua

```diff
@@ -27,30 +27,59 @@ function M.define_modules(...)
   configs.define_modules(...)
 end
 
-function M.statusline(indicator_size)
+local get_line_for_node = function(node, type_patterns, transform_fn)
+  local node_type = node:type()
+  local is_valid = false
+  for _, rgx in ipairs(type_patterns) do
+    if node_type:find(rgx) then
+      is_valid = true
+      break
+    end
+  end
+  if not is_valid then return '' end
+  local range = {node:range()}
+  local line = transform_fn(vim.trim(vim.fn.getline(range[1] + 1)))
```

Comment:
I think you should use https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/ts_utils.lua#L12-L12 to get the correct text from the node, otherwise you'll get too much noise from things that are in the same line.

Reply:
You're right, it is a bit cleaner. I updated it to use that function.